src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1289,6 +1296,20 @@ def forward(self, inputs, targets):
         else:
             raise RuntimeError("warmup_mode has to be either a name of a mode (str) or a subclass of PhaseCallback")
 
+        if isinstance(self.training_params.optimizer, str) or (
+            inspect.isclass(self.training_params.optimizer) and issubclass(self.training_params.optimizer, torch.optim.Optimizer)
+        ):
+            self.optimizer = build_optimizer(net=unwrap_model(self.net), lr=self.training_params.initial_lr, training_params=self.training_params)
+        elif isinstance(self.training_params.optimizer, torch.optim.Optimizer):
+            if self.training_params.initial_lr is not None:
+                raise RuntimeError("An instantiated optimizer cannot be passed along initial_lr != None")
+            self.optimizer = self.training_params.optimizer
+
+            # NEED TO EXTRACT INITAL_LR FROM THE OPTIMIZER PARAM GROUPS
+            self.training_params.initial_lr = get_initial_lr_from_optimizer(self.optimizer)
+        else:
+            raise UnsupportedOptimizerFormat()
```

Comment:
I think it would be very helpful to print some model statistics (similarly to what we print for dataset before we start training):
* Total number and number of optimizable parameters of the model
* A learning rate (If it's a single-value just print this value, if it's per-layer print it in detailed format.


Examples:

```
Model:         YoloNAS_S (34.2M parameters)
Learning rate: 1e-5
```


```
Model:          YoloNAS_S (34.2M parameters, 34.0M optimized)
Learning rates: 
  - default: 1e-5 (31M parameters)
  - head:    1e-4 (3M parameters)
```